### DISCUSSION TITLE
What is the alternative of `DataTransformer`?

### DISCUSSION THREAD

**Discussion by bikalbasnet** [api-platform/core#4892] [Q&A] [answered]
DataTransformer has been deprecated since `2.7.0-alpha.2` and removed in `3.0.0-beta.1`.

It says 
```
The DataTransformer pattern is deprecated, use a Provider or a Processor and either use your input or return a new output there.
```
and
```
DataTransformers: concept got removed, input and output classes are handled as anonymous resources 
```

I am trying to understand what is the reason behind this decision and can any one help me point out in the right direction of alternative way to do it when using DTO? 

## Reference: 
- https://github.com/api-platform/core/pull/4722
- https://github.com/api-platform/core/blob/main/CHANGELOG.md#300-beta1

**Comment by alanpoulain** [comment]
I think this choice has been made in order to simplify the usage of DTO and the underlying code in API Platform.
The transformation needs to be done inside your provider (or processor): instead of returning the resource, your return the corresponding DTO.

**Comment by soyuka** [comment, marked as answer]
It's become easier please check the documentation at https://api-platform.com/docs/main/core/dto/ (main branch). Basically all you need to do instead is a State Processor, you'll receive the input and process it as you like. If you still want to persist your data decorate our state process to do it magically. 

Let me know if something is unclear.
